src/mame/drivers/sandscrp.cpp

```diff
@@ -246,32 +251,31 @@ WRITE16_MEMBER(sandscrp_state::coincounter_w)
 
 READ16_MEMBER(sandscrp_state::latchstatus_word_r)
 {
-	return  (m_latch1_full ? 0x80 : 0) |
-			(m_latch2_full ? 0x40 : 0) ;
+	return  (m_latch_full[0] ? 0x80 : 0) |
+			(m_latch_full[1] ? 0x40 : 0) ;
 }
 
 WRITE16_MEMBER(sandscrp_state::latchstatus_word_w)
 {
 	if (ACCESSING_BITS_0_7)
 	{
-		m_latch1_full = data & 0x80;
-		m_latch2_full = data & 0x40;
+		m_latch_full[0] = data & 0x80;
+		m_latch_full[1] = data & 0x40;
 	}
 }
 
 READ16_MEMBER(sandscrp_state::soundlatch_word_r)
 {
-	m_latch2_full = 0;
-	return m_soundlatch2->read(space,0);
+	m_latch_full[1] = 0;
+	return m_soundlatch[1]->read(space,0);
 }
 
 WRITE16_MEMBER(sandscrp_state::soundlatch_word_w)
 {
 	if (ACCESSING_BITS_0_7)
 	{
-		m_latch1_full = 1;
-		m_soundlatch->write(space, 0, data & 0xff);
-		m_audiocpu->set_input_line(INPUT_LINE_NMI, PULSE_LINE);
+		m_latch_full[0] = 1;
+		m_soundlatch[0]->write(space, 0, data & 0xff);
 		m_maincpu->spin_until_time(attotime::from_usec(100)); // Allow the other cpu to reply
```

Comment:
This spin looks like a hack - is there any evidence that the main CPU is halted while the command is processed?  Is it working around an issue caused by the fact that the sound latch write goes through a schedule synchronisation barrier but `m_latch_full` is set immediately?

Reply:
Sounds are actually work after remove this hack, too.
I removed this hack for this aren't helpful for playing sound, And main CPU isn't seems like halting when writing latch.